### ISSUE TITLE
`no-useless-template-literals`: using `String` instead of template literal widens the type of the expression to just `string`

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3537]
Template literal -> `String(...)` conversion is not always equivalent and desirable. This code is reported:

```ts
const code = `${Math.random() > 0.5 ? 1 : 0}` as const; // Type: `"1" | "0"`
```

This is what the rule suggests as an alternative, but it breaks the expression type:

```ts
const code = String(Math.random() > 0.5 ? 1 : 0); // Type: `string`
```

TS playground for all the examples above: https://www.typescriptlang.org/play/?#code/MYewdgzgLgBKAmBTGBeGADAJAbwLIEMoALAOgCd8x4QBbACgEoYA+GABhIFYYB+GARhgAudgF90MfBDjhoAbgBQMZTAD0qmAD0eChaEiwEiAKpgoATwAOieKhgBlKGQCWYAOZ0CxcpWr0mrBzcfIIibAyKKmoa2kA

(If also actually keeps the `as const` part which causes a hard TS error `A 'const' assertion can only be applied to references to enum members, or string, number, boolean, array, or object literals`)

**Comment by hiSandog** [comment]
Could the exact `no-useless-template-literals` path from “'no-useless-template-literals': using 'String' instead of template literal widens the type of the expression to jus…” be retained in the test suite when this is fixed? The report already isolates a useful boundary, and asserting its expected result would help prevent recurrence.
